compiler/aarch64/codegen/FPTreeEvaluator.cpp

```diff
@@ -739,6 +739,21 @@ static TR::Register *fcmpHelper(TR::Node *node, TR::ARM64ConditionCode cc, bool
       }
 
    generateCSetInstruction(cg, node, trgReg, cc);
+   if (needsExplicitUnorderedCheck)
+      {
+      TR::Register *tmpReg = cg->allocateRegister();
+      if (cc == TR::CC_NE)
+         {
+         generateCSetInstruction(cg, node, tmpReg, TR::CC_VC);
+         generateTrg1Src2Instruction(cg, TR::InstOpCode::andx, node, trgReg, trgReg, tmpReg);
+         }
+      else
+         {
+         generateCSetInstruction(cg, node, tmpReg, TR::CC_VS);
+         generateTrg1Src2Instruction(cg, TR::InstOpCode::orrx, node, trgReg, trgReg, tmpReg);
+         }
```

Comment:
To me it would seem better to just choose the instruction opcodes in the if/else and then have the two calls to generate the instructions (rather than 4 calls overall).

Reply:
Updated as suggested.